mpes/fprocessing.py

```diff
@@ -39,9 +39,176 @@
 from functools import reduce
 from funcy import project
 from threadpoolctl import threadpool_limits
+import datetime as dt
+import urllib
+import json
+import xarray as xr
+import copy
 
 N_CPU = ps.cpu_count()
 
+aperture_dict = {
+    "2018-01-23T19:35:15":{
+        "fa_size":{750.:((-3.0, -1.4), (-5.4, -4.6)),
+                   "grid":((-3.0, -1.4), (0.15, 1.75)),
+                   1500.:((-3.0, -1.4), (6.25, 7.75)),
+                   200.:((3.3, 4.4), (-5.4, -4.6)),
+                   500.:((3.3, 4.4), (0.15, 1.75)),
+                   1000.:((3.3, 4.4), (6.25, 7.75)),
+                   20.:((9.6, 10.1), (-5.4, -4.6)),
+                   50.:((9.6, 10.1), (0.15, 1.75)),
+                   100.:((9.6, 10.1), (6.25, 7.75)),
+                   "open":((-10.4, -9.4), (-9.5, -8.9))},
+        "ca_size":{50.:(8.0, 8.4), 200.:(-0.5, -0.9),
+                   100:(3.4, 3.8), "grid":(-5.3, -5.9),
+                   "open":(-12.0, -10.8)}},
+    "2020-01-23T19:35:15":{
+        "fa_size":{750.:((-4.8, -6.2), (5.0, 6.0)),
+                  "grid":((-4.8, -6.2), (-0.3, -0.7)),
+                  500.:((-4.8, -6.2), (-6.0, -7.0)),
+                  200.:((0.5, 0.9), (-0.3, -0.7)),
+                  100.:((0.5, 0.9), (-6.0, -7.0)),
+                  300.:((0.5, 0.9), (5.0, 6.0)),
+                  10.:((6.5, 6.9), (-6.0, -7.0)),  
+                  20.:((6.5, 6.9), (-0.3, -0.7)),
+                  50.:((6.5, 6.9), (5.0, 6.0)),
+                  "open":((-10.4, -9.4), (-9.5, -8.9))},
+        "ca_size":{50.:(9.0, 11.0), 300.:(-0.1, 0.1),
+                   200.:(0.7, 1.5), 100:(5.1, 5.9),
+                   "grid":(-5.5, -5.2), "open":(-11.2, -10.8)}}
+    }

```

Comment:
The way you parse this dict, the left value must always be smaller than the right value. This is not the case for some of the negative values! i.e. -0.3 < f_hor  < -0.7 can never be true!